pkg/ee/default-application-controller/namespace.go

```diff
@@ -0,0 +1,39 @@
+/*
+                  Kubermatic Enterprise Read-Only License
+                         Version 1.0 ("KERO-1.0”)
+                     Copyright © 2024 Kubermatic GmbH
+
+   1.	You may only view, read and display for studying purposes the source
+      code of the software licensed under this license, and, to the extent
+      explicitly provided under this license, the binary code.
+   2.	Any use of the software which exceeds the foregoing right, including,
+      without limitation, its execution, compilation, copying, modification
+      and distribution, is expressly prohibited.
+   3.	THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package defaultapplicationcontroller
+
+import (
+	"context"
+
+	appskubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/apps.kubermatic/v1"
+)
+
+func GetAppNamespace(ctx context.Context, application *appskubermaticv1.ApplicationDefinition) *appskubermaticv1.AppNamespaceSpec {
+	if application.Spec.DefaultNamespace != nil {
+		return application.Spec.DefaultNamespace
+	}
+	return &appskubermaticv1.AppNamespaceSpec{
+		Name:   application.Name,
+		Create: true,
+	}
+}
```

Comment:
Why is this an EE only feature? Having a default namespace here is essential for some applications to work that require a specific namespace name for the applications to function properly. Making this EE only is not something I'm happy about but if this was already discussed and decided anywhere i'd like to know.

Reply:
The field https://github.com/kubermatic/kubermatic/blob/main/pkg/apis/apps.kubermatic/v1/application_definition.go#L223-L225 itself is not marked as EE anyways.

Reply:
As discussed internally, this will be generally available and not an EE only feature.